src/Workspaces/CSharp/Portable/Formatting/Rules/NewLineUserSettingFormattingRule.cs

```diff
@@ -159,6 +159,15 @@ public override AdjustSpacesOperation GetAdjustSpacesOperation(SyntaxToken previ
                 }
             }
 
+            // * { - in the switch expression context
+            if (currentToken.IsKind(SyntaxKind.OpenBraceToken) && currentToken.Parent.IsKind(SyntaxKind.SwitchExpression))
+            {
+                if (!optionSet.GetOption(CSharpFormattingOptions.NewLinesForBracesInControlBlocks))
```

Comment:
❗️  See https://github.com/dotnet/roslyn/issues/36913#issuecomment-546015089 for the design review conclusion here.

```suggestion
                if (!optionSet.GetOption(CSharpFormattingOptions.NewLinesForBracesInObjectCollectionArrayInitializers))
```